cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -116,6 +117,16 @@ bool SymbolList::can_update_symbol_list(const std::shared_ptr<Store>& store,
         store->iterate_type(KeyType::VERSION_REF, [&] (const auto& key) {
             auto id = variant_key_id(key);
             stream_ids.push_back(id);
+
+            if (stream_ids.size() == this->max_delta_ * 2 && !warned_expected_slowdown_) {
```

Comment:
What's the intent of the `warned_expected_slowdown_` for this one?
Why do we say "to resolve" as if some user action is required, this is already in the context of a `list_symbols` call isn't it so it's about to compact the cache anyway?

Reply:
> isn't it so it's about to compact the cache anyway?

You're right, it is. However this first run of `list_symbols` might take a _really_ long time depending on how many symbols there are. Currently it appears as if it is hanging if there are a lot - so even though we're about to resolve the issue it's still worth highlighting this to the user. 

It's also possible that this won't proceed to compaction anyway though because:

* The user may not have write privileges  
* The process may be killed prior to the compaction key being written

I'm definitely open to claifying the key points in the wording. It's hard to be clear without overloading!